firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -318,6 +314,19 @@ private InstallationTokenResult readGenerateAuthTokenResponse(HttpsURLConnection
     return builder.build();
   }
 
+  // Read the error message from the response.
+  private String readErrorResponse(HttpsURLConnection conn) throws IOException {
+    BufferedReader reader =
+        new BufferedReader(new InputStreamReader(conn.getErrorStream(), Charset.defaultCharset()));
+    StringBuilder response = new StringBuilder();
+    for (String input = reader.readLine(); input != null; input = reader.readLine()) {
+      response.append(input);
+    }
```

Comment:
Hmm, based off my reading of the docs BufferedReader#readLine will return a string (or null) with no newlines, so we're turning this:

```
foo
bar
baz
```
into this:
```
foobarbaz
```

Is that intended?

Reply:
Good catch. Thanks. Updated. PTAL.

Reply:
I wouldn't bother with the system-dependent newline, just use '\n' (Android will never switch this and it's clearer and slightly more efficient)

Reply:
Done.